src/core/obj.js

```diff
@@ -135,6 +135,37 @@ var Dict = (function DictClosure() {
       return Promise.resolve(value);
     },
 
+    // Same as get(), but dereferences all elements if the result is an Array.
+    getArray: function Dict_getArray(key1, key2, key3) {
+      var value = this.get(key1, key2, key3);
+      var xref = this.xref;
+      if (!isArray(value)) {
+        return value;
+      }
+      var valueLen = value.length, valueArray = new Array(valueLen);
+      for (var i = 0; i < valueLen; i++) {
+        valueArray[i] = xref.fetchIfRef(value[i]);
+      }
+      return valueArray;
+    },
+
+    // Same as getAsync(),
+    // but dereferences all elements if the result is an Array.
+    getAsyncArray: function Dict_getAsyncArray(key1, key2, key3) {
```

Comment:
> We can remove this function: not used

Done. (I just added it for consistency with the other method, but since it's not used I agree with removing it.)

> [...] and fetchIfRefs inside probably need to be async fetched as well.

This is an embarrassing oversight on my part :-)